src/mono/mono/metadata/assembly.c

```diff
@@ -3188,6 +3204,164 @@ mono_register_bundled_satellite_assemblies (const MonoBundledSatelliteAssembly *
 	satellite_bundles = assemblies;
 }
 
+static GHashTable *bundled_resources = NULL;
```

Comment:
`assembly.c` is pretty big already. does the bundled resource stuff really belong in here.  Maybe ou could add some new `mono/metadata/bundled-resources.c` and `bundled-resources.h` (`bundled-resources-internals.h` ?) for this stuff.

It would also help to define a crisper interface between the bundling logic and the rest of the assembly loading logic.